frontend/src/resources/userpreference.tsx

```diff
@@ -0,0 +1,90 @@
+/* Copyright Contributors to the Open Cluster Management project */
+import { getUsername } from '../lib/username'
+import { Metadata } from './metadata'
+import { IResource, IResourceDefinition } from './resource'
+import { createResource, patchResource } from './utils/resource-request'
+
+export const UserPreferenceApiVersion = 'console.open-cluster-management.io/v1'
+export type UserPreferenceApiVersionType = 'console.open-cluster-management.io/v1'
+
+export const UserPreferenceKind = 'UserPreference'
+export type UserPreferenceKindType = 'UserPreference'
+
+export const UserPreferenceDefinition: IResourceDefinition = {
+    apiVersion: UserPreferenceApiVersion,
+    kind: UserPreferenceKind,
+}
+
+export interface UserPreference extends IResource {
+    apiVersion: UserPreferenceApiVersionType
+    kind: UserPreferenceKindType
+    metadata: Metadata
+    spec?: {
+        savedSearches?: SavedSearch[]
+    }
+}
+
+export interface SavedSearch {
+    description?: string
+    id: string
+    name: string
+    searchText: string
+}
+
+export function createUserPreference(savedSearches: SavedSearch[]) {
+    return getUsername()
+        .promise.then((payload) => {
+            const username =
+                payload && payload.body && payload.body.username
+                    ? payload.body.username.toLowerCase().replace(/[^a-z0-9-.]/g, '-')
+                    : 'undefined'
+            createResource<UserPreference>({
+                apiVersion: UserPreferenceApiVersion,
+                kind: UserPreferenceKind,
+                metadata: {
+                    name: username,
+                },
+                spec: {
+                    savedSearches,
+                },
+            })
+        })
+        .catch((error) => {
+            // eslint-disable-next-line no-console
+            console.error(error)
```

Comment:
Did you intend to keep these?

Reply:
I was planning to include them yes. And I will comb through the code smells - thanks.